File: voting-backend/src/main/java/ch/rafaelurben/tamtour/voting/dto/VotingCategoryUserDetailDto.java
package ch.rafaelurben.tamtour.voting.dto;

import java.util.Collection;

public record VotingCategoryUserDetailDto(
    VotingCategoryBaseDto category,
    VotingCategoryUserStateDto userState,
    Collection<VotingCandidateDto> candidates,
    VotingPositionMapDto positionMap) {}

File: voting-frontend/src/app/dto/voting-category-user-detail.dto.ts
import { VotingCategoryBaseDto } from './voting-category-base.dto';
import { VotingCategoryUserStateDto } from './voting-category-user-state.dto';
import { VotingCandidateDto } from './voting-candidate.dto';
import { VotingPositionMapDto } from './voting-position-map.dto';

export interface VotingCategoryUserDetailDto {
  category: VotingCategoryBaseDto;
  userState: VotingCategoryUserStateDto;
  candidates: VotingCandidateDto[];
  positionMap: VotingPositionMapDto;
}
